baseapp/abci.go

```diff
@@ -53,21 +53,32 @@ func (app *BaseApp) InitChain(req abci.RequestInitChain) (res abci.ResponseInitC
 	app.setDeliverState(initHeader)
 	app.setCheckState(initHeader)
 
+	// add block gas meter for any genesis transactions (allow infinite gas)
+	app.deliverState.ctx = app.deliverState.ctx.WithBlockGasMeter(sdk.NewInfiniteGasMeter())
+
+	// wrap in a cache context for state listening to work.
+	cacheCtx, write := app.deliverState.ctx.CacheContextWithListeners(app.writeListeners)
```

Comment:
I don't understand why we need this? Why do we need to cache wrap in order to have listening work?